generators/aws-containers/aws-client.js

```diff
@@ -0,0 +1,299 @@
+ * the bucketBaseName will be used to create a
+ * @param bucketName the name of the bucket to create.
+ * @param region the region to create the bucket in.
+ * @returns {Promise}
+ */
+function createS3Bucket(bucketName, region = DEFAULT_REGION) {
+    const createBuckerParams = {
+        Bucket: bucketName
+    };
+    return spinner(new Promise((resolve, reject) => s3.headBucket({
+        Bucket: bucketName
+    }).promise()
+        .catch((error) => {
+            if (error.code !== 'NotFound') {
+                reject(new Error(`The S3 Bucket ${chalk.bold(bucketName)} in region ${chalk.bold(region)} already exists and you don't have access to it. Error code: ${chalk.bold(error.code)}`));
+            }
+        })
+        .then(() =>
+            s3.createBucket(createBuckerParams).promise()
+                .then(resolve)
+                .catch(error => reject(new Error(`There was an error during the creation of the S3 Bucket ${chalk.bold(bucketName)} in region ${chalk.bold(region)}`))))));
+}
+
+/**
+ * Upload the template in the S3Bucket
+ * @param bucketName S3 Bucket name to upload the template into
+ * @param filename Name to give to the file in the Bucket
+ * @param path Path to the file
+ * @returns {Promise}
+ */
+function uploadTemplate(bucketName, filename, path) {
+    return spinner(new Promise((resolve, reject) =>
+        fs.stat(path, (error, stats) => {
+            if (_.isNil(stats)) {
+                reject(new Error(`File ${chalk.bold(path)} not found`));
+            }
+            const upload = s3.upload(
+                {
+                    Bucket: bucketName,
+                    Key: filename,
+                    Body: fs.createReadStream(path)
+                },
+                {
+                    partSize: Math.max(stats.size, S3_MIN_PART_SIZE),
+                    queueSize: 1
+                }
+            );
+            let bar;
+            upload.on('httpUploadProgress', (evt) => {
+                if (bar === undefined && evt.total) {
+                    const total = evt.total / 1000000;
+                    bar = new ProgressBar('uploading [:bar] :percent :etas', {
+                        complete: '=',
+                        incomplete: ' ',
+                        width: 20,
+                        total,
+                        clear: true
+                    });
+                }
+
+                const curr = evt.loaded / 1000000;
+                bar.tick(curr - bar.curr);
+            });
+            return upload.promise()
+                .then(resolve)
+                .catch(reject);
+        })));
+}
+
+/**
+ * Sanitize the bucketName following the rule found here:
+ * http://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-s3-bucket-naming-requirements.html
+ * @param bucketName
+ * @returns {string}
+ */
+function sanitizeBucketName(bucketName) {
+    let labels = _.split(bucketName, '.');
```

Comment:
Two things:
  - Instead of using lodash, you can use the regular API
  - You can chain the calls